activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -55,36 +55,42 @@ def not(opts, *rest)
         @scope
       end
 
-      # Returns a new relation with joins and where clause to identify
-      # associated relations.
+      # Returns a new relation that makes possible to identify associated relations.
       #
       # For example, posts that are associated to a related author:
       #
       #    Post.where.associated(:author)
       #    # SELECT "posts".* FROM "posts"
       #    # INNER JOIN "authors" ON "authors"."id" = "posts"."author_id"
-      #    # WHERE "authors"."id" IS NOT NULL
       #
-      # Additionally, multiple relations can be combined. This will return posts
+      # Multiple associations can be combined. This will return posts
       # associated to both an author and any comments:
       #
       #    Post.where.associated(:author, :comments)
       #    # SELECT "posts".* FROM "posts"
       #    # INNER JOIN "authors" ON "authors"."id" = "posts"."author_id"
       #    # INNER JOIN "comments" ON "comments"."post_id" = "posts"."id"
-      #    # WHERE "authors"."id" IS NOT NULL AND "comments"."id" IS NOT NULL
+      #
+      # You can use a relation in order to get an exists clause.
+      # This will return posts that are associated to a related author:
+      #
+      #    Post.where.associated(Author.where("authors.id = posts.author_id"))
+      #    # SELECT "posts".* FROM "posts"
+      #    # WHERE EXISTS (SELECT 1 FROM "authors" WHERE (authors.id = posts.author_id))
       def associated(*associations)
         associations.each do |association|
-          reflection = scope_association_reflection(association)
-          @scope.joins!(association)
-          self.not(association => { reflection.association_primary_key => nil })
```

Comment:
For `where.associated`, I believe `author.id` will never be null, so just a join is enough to find associated records.